app/routes.py

```diff
@@ -1,2 +1,217 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from datetime import datetime
+from flask import request, Blueprint, make_response, jsonify
+import requests
+import os
+# from slack_sdk import WebClient
+# from slack_sdk.errors import SlackApiError
+from app.models.goal import Goal
 
+
+goals_bp = Blueprint(
+    "goals", __name__, url_prefix="/goals")
+tasks_bp = Blueprint(
+    "tasks", __name__, url_prefix="/tasks")
+
+
+# -------------------------
+# WAVE 1 - TASK ENDPOINTS
+# -------------------------
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_task():
+    request_body = request.get_json()
+    if "title" in request_body and "description" in request_body and "completed_at" in request_body:
+        new_task = Task(
+            title=request_body["title"],
+            description=request_body["description"],
+            completed_at=request_body["completed_at"]
+        )
+        db.session.add(new_task)
+        db.session.commit()
+        return jsonify({"task": new_task.to_json()}), 201
+    else:
+        return make_response({"details": "Invalid data"}, 400)
+
+
+# WAVE 2
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def task_index():
+    sort_query = request.args.get("sort")
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title)
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+    tasks_response = [(task.to_json()) for task in tasks]
+    return make_response(jsonify(tasks_response), 200)
+
+
+# WAVE 1
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def get_one_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
```

Comment:
we could shorten this by using `get_or_404()` method instead

```suggestion
    task = Task.query.get_or_404(task_id)
```